.github/workflows/npm.yml

```diff
@@ -0,0 +1,72 @@
+# Releases a canary release when new commits merge to main
+# Releases an official release when workflow dispactch is
+# manually triggered
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+name: NPM Build 📦
+
+jobs:
+  build:
+    name: Publish to NPM 📦
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+          cache: "yarn"
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Install dependencies
+        run: yarn install --network-concurrency 1
+
+      - name: Build and install CLI
+        working-directory: packages/cli
+        run: yarn build && yarn link
+
+      - name: Generate tables
+        run: yarn tablegen
+
+      - name: Outdated files detected, run `yarn tablegen` and commit them
+        run: |
+          if [[ -n "$(git status --porcelain)" ]]; then
+            git status
+            git --no-pager diff
+            exit 1
+          fi
+
+      - name: Generate gas reports
+        run: yarn gas-report
+
+      - name: Outdated files detected, run `yarn gas-report` and commit them
+        run: |
+          if [[ -n "$(git status --porcelain)" ]]; then
+            git status
+            git --no-pager diff
+            exit 1
+          fi
+
+      - name: Publishing canary releases to NPM 📦
+        if: github.event_name != 'workflow_dispatch'
+        run: yarn release:canary
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Publishing new release to NPM 📦
+        if: github.event_name == 'workflow_dispatch'
+        run: yarn release
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
I guessed this is the env variable we need to set in the github actions secrets.   I am not 100% sure lerna is expecting NPM_TOKEN